### DISCUSSION TITLE
RemovalListener Internal working for callback on expireAfterCreate

### DISCUSSION THREAD

**Discussion by pushkardhat** [ben-manes/caffeine#493] [Q&A] [answered]
@ben-manes 
We have a scenario where in we want to invoke some code on expireAfterCreate for each entry in the cache using onRemoval.
On Expiry of an entry, onRemoval is not being called, unless put() or get() operation is performed on cache.
Is there a way by which we can invoke RemovalListener on each expired entry of cache?

**Comment by ben-manes** [comment, marked as answer]
It sounds like you want to use a scheduler to evict the entry when no other activity on the cache is occurring. Please see this [wiki page](https://github.com/ben-manes/caffeine/wiki/Cleanup) for more details.

```java
LoadingCache<Key, Graph> graphs = Caffeine.newBuilder()
    .scheduler(Scheduler.systemScheduler())
    .expireAfterWrite(10, TimeUnit.MINUTES)
    .build(key -> createExpensiveGraph(key));
```

The `Scheduler.systemScheduler()` is available in Java 9+ and uses a JVM provided thread, or else you can provide a `ScheduledExecutorService`. The cache will peek at the next entry to expire and schedule a `cleanUp()` call.

**Reply by pushkardhat** [reply]
Do we need to implement Runnable for ScheduledExecutorService will cleanup() call.
Or below will handle it on its own?
executor = Executors.newSingleThreadScheduledExecutor(); 
scheduler(Scheduler.forScheduledExecutorService(executor))

**Reply by ben-manes** [reply]
That will handle it. The cache will supply the runnable, etc.
